src/ui/Facet/Facet.ts

```diff
@@ -1528,6 +1528,35 @@ export class Facet extends Component {
     this.dependsOnManager = new DependsOnManager(facetInfo);
   }
 
+  private dependsOnUpdateParentDisplayValue() {
+    if (this.options.dependsOn) {
+      const masterFacetHTML = $$(this.root)
+        .findAll(`.CoveoFacet`)
+        .filter(HTMLFacet => {
+          const facet = <Facet>Facet.get(HTMLFacet);
+          return facet.options.id === this.options.dependsOn;
+        });
+      if (!masterFacetHTML.length) {
+        this.logger.warn(
+          `Unable to find a Facet with the id or field "${this.options.dependsOn}".`,
+          `The master Facet values can't be updated.`
+        );
+        return;
+      }
+      if (masterFacetHTML.length > 1) {
+        this.logger.warn(
+          `Mulltiple Facet with the id or field "${this.options.dependsOn}" found.`,
+          `A Facet can only depends on one other Facet.`,
+          `The master Facet values can't be updated.`,
+          masterFacetHTML
+        );
+        return;
```

Comment:
Would it be easy to add a unit test for behaviour of not updating the master facet when there are 2 facets with the same id?

Reply:
The only way I see I can be certain is to spy on the console and check the log. I'll try that. If you have any other idea, feel free to share them

Reply:
What about checking if the `masterFacet.keepDisplayedValuesNextTime` was set to `false` or not?

If you spy on the log, that could work if you check `toHaveBeenCalledWith` and the exact message, but it's more indirect of a check For example, if someone removes the `return` on line 1553, the test would still pass even though it should not.

Reply:
I tried the console, it worked locally but not when running the specrunner but not with `npm run test`  
Finally, I did as you suggested, I forgot that it was public.